query-planner-js/src/__tests__/buildPlan.test.ts

```diff
@@ -7973,3 +7973,195 @@ describe('@requires references external field indirectly', () => {
     `);
   });
 });
+
+describe('handles fragments with directive conditions', () => {
+  test('fragment with interseting parent type and directive condition', () => {
+    const subgraphA = {
+      typeDefs: gql`
+        directive @test on FRAGMENT_SPREAD
+        type Query {
+          i: I
+        }
+        interface I {
+          _id: ID
+        }
+        type T1 implements I @key(fields: "id") {
+          _id: ID
+          id: ID
+        }
+        type T2 implements I @key(fields: "id") {
+          _id: ID
+          id: ID
+        }
+      `,
+      name: 'A',
+    };
+
+    const subgraphB = {
+      typeDefs: gql`
+        directive @test on FRAGMENT_SPREAD
+        type Query {
+          i2s: [I2]
+        }
+        interface I2 {
+          id: ID
+          title: String
+        }
+        type T1 implements I2 @key(fields: "id") {
+          id: ID
+          title: String
+        }
+        type T2 implements I2 @key(fields: "id") {
+          id: ID
+          title: String
+        }
+      `,
+      name: 'B',
+    };
+
+    const [api, queryPlanner] = composeAndCreatePlanner(subgraphA, subgraphB);
+
+    const operation = operationFromDocument(
+      api,
+      gql`
+        query {
+          i {
+            _id
+            ... on I2 @test {
+              id
+            }
+          }
+        }
+      `,
+    );
+
+    expect(operation.expandAllFragments().toString()).toMatchInlineSnapshot(`
```

Comment:
Was this line previously failing? It seems superfluous to me, but I wouldn't take it out, just curious.

Reply:
this is just a copy and paste -> when looking into the reported issue I started with the second test and this got copied over. Operation normalization was fine, we were just missing the directives in the query plan. I'll remove this.